hack/quickstart-setup.sh

```diff
@@ -0,0 +1,322 @@
+metadata:
+  name: ${KIND_CLUSTER_PREFIX}aws-credentials
+  namespace: multi-cluster-gateways
+type: "kuadrant.io/aws"
+stringData:
+  AWS_ACCESS_KEY_ID: ${MGC_AWS_ACCESS_KEY_ID}
+  AWS_SECRET_ACCESS_KEY: ${MGC_AWS_SECRET_ACCESS_KEY}
+  AWS_REGION: ${MGC_AWS_REGION}
+EOF
+
+    cat <<EOF | kubectl apply -f -
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: ${KIND_CLUSTER_PREFIX}controller-config
+  namespace: multi-cluster-gateways
+data:
+  AWS_DNS_PUBLIC_ZONE_ID: ${MGC_AWS_DNS_PUBLIC_ZONE_ID}
+  ZONE_ROOT_DOMAIN: ${MGC_ZONE_ROOT_DOMAIN}
+  LOG_LEVEL: "${LOG_LEVEL}"
+EOF
+
+    cat <<EOF | kubectl apply -f -
+apiVersion: kuadrant.io/v1alpha1
+kind: ManagedZone
+metadata:
+  name: ${KIND_CLUSTER_PREFIX}dev-mz
+  namespace: multi-cluster-gateways
+spec:
+  id: ${MGC_AWS_DNS_PUBLIC_ZONE_ID}
+  domainName: ${MGC_ZONE_ROOT_DOMAIN}
+  description: "Dev Managed Zone"
+  dnsProviderSecretRef:
+    name: ${KIND_CLUSTER_PREFIX}aws-credentials
+    namespace: multi-cluster-gateways
+    type: AWS
+EOF
+}
+
+deployQuickStartHub() {
+    clusterName=${1}
+    kubectl config use-context kind-${clusterName}
+    echo "Initialize quickstart setup on ${clusterName}"
+    kustomize build ${QUICK_START_HUB_KUSTOMIZATION} --enable-helm --helm-command helm | kubectl apply -f -
+    echo "Waiting for ingress-nginx deployments to be ready"
+    kubectl -n ingress-nginx wait --timeout=600s --for=condition=Available deployments --all
+    echo "Waiting for metallb-system deployments to be ready"
+    kubectl -n metallb-system wait --for=condition=ready pod --selector=app=metallb --timeout=300s
+    echo "Waiting for external-dns deployments to be ready"
+    kubectl -n external-dns wait --timeout=300s --for=condition=Available deployments --all
+    echo "Waiting for cert-manager deployments to be ready"
+    kubectl -n cert-manager wait --timeout=300s --for=condition=Available deployments --all
+}
+
+deployQuickStartSpoke() {
+   clusterName=${1}
+    kubectl config use-context kind-${clusterName}
+    echo "Initialize quickstart setup on ${clusterName}"
+    kustomize build ${QUICK_START_SPOKE_KUSTOMIZATION} --enable-helm --helm-command helm | kubectl apply -f -
+    echo "Waiting for ingress-nginx deployments to be ready"
+    kubectl -n ingress-nginx wait --timeout=600s --for=condition=Available deployments --all
+    echo "Waiting for metallb-system deployments to be ready"
+    kubectl -n metallb-system wait --for=condition=ready pod --selector=app=metallb --timeout=300s
+}
+
+configureControlCluster() {
+
+    clusterName=${1}
+    # Ensure the current context points to the control plane cluster
+    kubectl config use-context kind-${clusterName}
+    kubectl label managedcluster kind-mgc-control-plane ingress-cluster=true
+    cat <<EOF | kubectl apply -f -
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: GatewayClass
+metadata:
+  name: kuadrant-multi-cluster-gateway-instance-per-cluster
+spec:
+  controllerName: "kuadrant.io/mgc-gw-controller"
+EOF
```

Comment:
This reminds me, you won't be able to add CRDs and use them in the same `kustomize` command, you either need two separate apply commands, one with the CRDS, and one with the stuff that uses it. Or you can do what i did here and just run the command again https://github.com/mikenairn/multi-cluster-traffic-controller/blob/56059eb0fe82fcf99247aa2fffcd68f07aef331d/hack/local-setup.sh#L175-L177. Might not be the nicest for a quick start though.

Reply:
yes i hit that with GatewayClass and one other that is escaping me memory now. It's a pity. Would be nice to apply the CRD and cr in one go. I'll go ahead and use the mechanism above because my preference was to get all yaml (excluding that that depends on an env var) into one of the two quickstart kustomizes. 👍